cogs/Announce.py

```diff
@@ -0,0 +1,418 @@
+        """
+        Check if a particular id has an active announcement pending announcement
+        :param guild_id: The id of the guild of the command
+        :return: Boolean of whether there is an active announcement or not
+        """
+        return guild_id in self.messages.keys()
+
+    def get_role_names(self, guild_id, roles):
+        """
+        A function to get the names of all the roles the announcement will be sent to
+        :param roles: The list of roles in the guild
+        :param guild_id: The id of the guild
+        :return: All the names of the roles that are tagged
+        """
+        temp = []
+        for role in self.roles[guild_id]:
+            temp.append(discord.utils.get(roles, id=role).name)
+        return temp
+
+    def get_receivers(self, guild_id, roles):
+        """
+        A function to get the receivers of a particular announcement
+        :param roles: The list of roles in the guild
+        :param guild_id: The id of the guild
+        :return: All the receivers of the announcement
+        """
+        temp = []
+        for role in self.roles[guild_id]:
+            temp += discord.utils.get(roles, id=role).members
+        return list(set(temp))
+
+    def receiver_msg(self, guild):
+        """
+        A function to create a string message about receivers
+        :param guild: The guild of the bot
+        :return: A string message about receivers
+        """
+        if not self.roles[guild.id]:
+            return f"You are currently sending to Everyone and there are {str(len(guild.members))} receivers"
+        return f"You are currently sending to {self.get_role_names(guild.id, guild.roles)} and there are {str(len(self.get_receivers(guild.id, guild.roles)))} receivers "
+
+    def construct_embed(self, guild: discord.Guild):
+        """
+        Constructing an embedded message from the information stored in the manager
+        :param guild: The the guild
+        :return: An embedded message for the announcement
+        """
+        message = self.messages[guild.id]
+        embed: discord.Embed = discord.Embed(title=message.title,
+                                             description=message.description, colour=KoalaColours.KOALA_GREEN)
+        embed.set_author(name="announcement from " + guild.name)
+        if message.thumbnail != 'https://cdn.discordapp.com/':
+            embed.set_thumbnail(url=message.thumbnail)
+        return embed
+
+    @commands.group(name="announce")
+    async def announce(self, ctx):
+        """
+        Use k!announce create to create an announcement
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.send(f"Please use `{KoalaBot.COMMAND_PREFIX}help announce` for more information")
+
+    @commands.check(announce_is_enabled)
+    @announce.command(name="create")
+    async def create(self, ctx):
+        """
+        This command creates a new message that will be available for sending
```

Comment:
```suggestion
        Creates a new message that will be available for sending
```
Remove 'this command' from all docstrings, otherwise it comes through for everyone when doing `k!help`
![image](https://user-images.githubusercontent.com/30408869/115018795-370d8e00-9eb0-11eb-9a4f-80c75bd8162e.png)